runtime/near-vm-runner/src/lib.rs

```diff
@@ -1,41 +1,34 @@
 #![doc = include_str!("../README.md")]
 
-#[cfg(test)]
-mod tests;
-
 mod cache;
 mod errors;
 mod imports;
 #[cfg(feature = "wasmer0_vm")]
 mod memory;
-mod vm_kind;
-
 mod preload;
 pub mod prepare;
 mod runner;
-
+#[cfg(test)]
+mod tests;
+mod vm_kind;
+#[cfg(feature = "wasmer2_vm")]
+mod wasmer2_runner;
 #[cfg(feature = "wasmer0_vm")]
 mod wasmer_runner;
-
 #[cfg(feature = "wasmtime_vm")]
 mod wasmtime_runner;
 
-#[cfg(feature = "wasmer2_vm")]
-mod wasmer2_runner;
-
 pub use near_vm_errors::VMError;
-pub use preload::{ContractCallPrepareRequest, ContractCallPrepareResult, ContractCaller};
-pub use runner::compile_module;
-pub use runner::run;
-
 pub use near_vm_logic::with_ext_cost_counter;
 
 pub use cache::get_contract_cache_key;
 pub use cache::precompile_contract;
 pub use cache::precompile_contract_vm;
 pub use cache::MockCompiledContractCache;
+pub use preload::{ContractCallPrepareRequest, ContractCallPrepareResult, ContractCaller};
+pub use runner::{run, Runtime};
 
-// These two are public for the standalone runner, but are an implementation
-// detail of `near-vm-runner`. Public API like `run` should not expose VMKind.
-pub use runner::run_vm;
-pub use vm_kind::VMKind;
+// This is public for internal experimentation use only, and should otherwise be considered an
+// implementation detail of `near-vm-runner`.
+#[doc(hidden)]
+pub use vm_kind::VMKind as VMKindPublicForInternalUseOnly;
```

Comment:
I think this is a bit over the top :-) We currently don't promise any kind of API stability, so, practically, I don't think we should go out of our way to discourage the use. 

What about just 

```rust
/// This is public for internal experimentation use only, and should otherwise be considered an
/// implementation detail of `near-vm-runner`.
pub mod internals {
  pub use vm_kind::VMKind;
}
```

?

Reply:
I can do that. My goal here was to document this for myself and the future readers who look at this for the first time pretty much. Up until very recently I thought this was intended to be a public API and either approach helps with understanding this. Though I think there is an opportunity to actually make this private still and depend on `with_vm_variants` to iterate through the available `Runtime` implementations.

Reply:
Let's go with `internal` module than. The specific think I don't like with the alias is that we rename at every call-site -- that sounds like an implicit rule for consistency, which will be broken very-very soon. :+1: to the overall idea of reflecting public API in code.

>Though I think there is an opportunity to actually make this private still and depend on with_vm_variants to iterate through the available Runtime implementations.

Not sure this'll work, it makes sense that estimator has `--vm-kind` CLI argument: 

https://github.com/near/nearcore/blob/cb20e72afeed2c4d6f6d06e87fd857c5438af8ce/runtime/runtime-params-estimator/src/main.rs#L49-L50